### DISCUSSION TITLE
Transcribe only a part of the audio?

### DISCUSSION THREAD

**Discussion by Abdurrafey-Siddiqui** [openai/whisper#755] [Q&A]
If an audio is like 50 minutes, is there a parameter through which i can only transcribe the first 20 mintutes? or even transcribe from the middle of the audio say 30:00-50:00 for example.

**Comment by jotixh** [comment]
I was trying to do the same thing a few days ago, but I could not find a time parameter for the function. So I made a function that cuts the file, creates a temporary one and transcribes it with whisper. 

I hope it works for you. 

```python
def cut_audio(fileaudio, start_seconds, end_seconds):
    #!pip3 install pydub
    from pydub import AudioSegment

    output = "tmp.wav"

    sound = AudioSegment.from_wav(fileaudio) # for mp3: AudioSegment.from_mp3()

    StrtTime = float(start_seconds) * 1000
    EndTime  = float(end_seconds) * 1000
    extract = sound[StrtTime:EndTime]

    # save
    extract.export(output, format="wav")

    return output

# cut file from 3 to second 8
fileout = cut_audio("myaudio.wav", 3, 8) 


### WHISPER
import whisper
whisper_model = whisper.load_model("medium")
result = whisper_model.transcribe(fileout, language = "en") 
print(result['text'])
```

**Comment by glangford** [comment]
This has also been discussed in
https://github.com/openai/whisper/discussions/135 - Running on selected time segments of an audio
https://github.com/openai/whisper/discussions/467 - Select or exclude audio parts before transcribing

**Reply by glangford** [reply]
I like the simplicity of the suggestion of @jongwook from https://github.com/openai/whisper/discussions/135

> This package doesn't support for this as-is, but you could manually create segmented files using ffmpeg, e.g.
> 
> ffmpeg -i input.mp3 -ss 2:20.00 -to 3:40.00 output.mp3
